UGW: Fixes 'toolchain/uClibc/utils/compile' in some machines while toolchain is installed in a different path.
This fix is to install toolchain in a common path by make tools/install; make toolchain/install TOOLCHAIN_DIR="<different path>"
Above commands works normally but fails in some latest distributions of Ubuntu and Fedora. Failure found in make toolchain/uClibc/utils/compile.
It says lgcc not found. Feed uses mips-linux-openwrt-gcc initial to compile uClibc utils ($(TOOLCHAIN_DIR)/initial/bin:$(TARGET_PATH)) where it fails to link lgcc.
This can be fixed by using gcc final. So used:- $(TOOLCHAIN_DIR)/bin:$(TARGET_PATH) which is gcc final.

diff --git a/toolchain/uClibc/common.mk b/toolchain/uClibc/common.mk
--- a/toolchain/uClibc/common.mk
+++ b/toolchain/uClibc/common.mk
@@ -61,6 +61,16 @@ UCLIBC_MAKE = PATH='$(TOOLCHAIN_DIR)/ini
 	LIBGCC="$(subst libgcc.a,libgcc_initial.a,$(shell $(TARGET_CC) -print-libgcc-file-name))" \
 	DOSTRIP=""
 
+UCLIBC_UTILS_MAKE = PATH='$(TOOLCHAIN_DIR)/bin:$(TARGET_PATH)' $(MAKE) -C $(HOST_BUILD_DIR) \
+	$(TARGET_CONFIGURE_OPTS) \
+	DEVEL_PREFIX=/ \
+	RUNTIME_PREFIX=/ \
+	HOSTCC="$(HOSTCC)" \
+	CPU_CFLAGS="$(CPU_CFLAGS)" \
+	ARCH="$(CONFIG_ARCH)" \
+	LIBGCC="$(subst libgcc.a,libgcc_initial.a,$(shell $(TARGET_CC) -print-libgcc-file-name))" \
+	DOSTRIP=""
+
 define Host/Prepare
 	$(call Host/Prepare/Default)
 	$(if $(strip $(QUILT)), \
diff --git a/toolchain/uClibc/utils/Makefile b/toolchain/uClibc/utils/Makefile
--- a/toolchain/uClibc/utils/Makefile
+++ b/toolchain/uClibc/utils/Makefile
@@ -7,7 +7,7 @@ HOST_STAMP_INSTALLED:=$(TOOLCHAIN_DIR)/s
 
 define Host/Compile
 	$(SED) 's,^CROSS=.*,CROSS=$(TARGET_CROSS),g' $(HOST_BUILD_DIR)/Rules.mak
-	$(UCLIBC_MAKE) PREFIX= utils
+	$(UCLIBC_UTILS_MAKE) PREFIX= utils
 endef
 
 define Host/Install
